include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -2503,6 +2508,106 @@ namespace internal
   }
 
 
+
+  /**
+   * Dummy class.
+   */
+  template <int dim,
+            int n_rows,
+            int n_columns,
+            typename Number,
+            typename Number2>
+  struct EvaluatorTensorProduct<evaluate_none,
+                                dim,
+                                n_rows,
+                                n_columns,
+                                Number,
+                                Number2>
+  {
+    static constexpr unsigned int n_rows_of_product =
+      Utilities::pow(n_rows, dim);
+    static constexpr unsigned int n_columns_of_product =
+      Utilities::pow(n_columns, dim);
+
+    EvaluatorTensorProduct()
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    EvaluatorTensorProduct(const AlignedVector<Number2> &,
+                           const AlignedVector<Number2> &,
+                           const AlignedVector<Number2> &,
+                           const unsigned int = 0,
+                           const unsigned int = 0)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    values(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    gradients(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    hessians(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    values_one_line(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    gradients_one_line(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int direction, bool contract_over_rows, bool add>
+    void
+    hessians_one_line(const Number[], Number[]) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int  direction,
+              bool contract_over_rows,
+              bool add,
+              bool one_line = false>
+    static void
+    apply(const Number2 *DEAL_II_RESTRICT, const Number *, Number *)
+    {
+      Assert(false, ExcNotImplemented());
+    }
+
+    template <int  face_direction,
+              bool contract_onto_face,
+              bool add,
+              int  max_derivative,
+              bool lex_faces = false>
+    void
+    apply_face(const Number *DEAL_II_RESTRICT, Number *DEAL_II_RESTRICT) const
+    {
+      Assert(false, ExcNotImplemented());
+    }
+  };
```

Comment:
here

Reply:
Do we ever intend to fill this class? If not, I see little value in providing it and would rather like to see an alternative that avoids those 100 lines, and suggest to either use one of the existing alternatives or do some template magic elsewhere to avoid calling the undefined functions.

Reply:
@kronbichler I have pushed an alternative version. It is nor really based on template magic but only on a template specialization ;)